src/tests/api/auth/sign-in/service.spec.ts

```
@@ -0,0 +1,108 @@
+import { service } from "api/auth/sign-in/service";
+import { tokenGenerator } from "utils/token/token-generator";
+import { CustomError } from "utils/error";
+import { StatusCodeEnum } from "enums/status-code";
+import { refreshTokenMock } from "tests/mocks/refresh-token";
+import { userMock } from "tests/mocks/user";
+
+describe("auth > sign-in > service", () => {
+	const validEmail = "johndoe@foo.foo";
+	const validPassword = "JohnDoe@123";
+	const invalidPassword = "John@123Doe";
+
+	describe("Success", () => {
+		it("should sign-in", async () => {
+			let result: any;
+
+			refreshTokenMock.create.mockResolvedValue(refreshTokenMock.doc());
+
+			const fakeUserId = "7ee0fb6a-ce98-4499-b4f5-5738eb6c4efd";
+			userMock.getByEmail.mockResolvedValue(
+				userMock.doc({ userId: fakeUserId }),
+			);
+
+			try {
+				result = await service(
+					{
+						createRefreshTokenRepository: refreshTokenMock.create,
+						getByEmailUserRepository: userMock.getByEmail,
+					},
+					{
+						email: validEmail,
+						password: validPassword,
+					},
+				);
+			} catch (err) {
+				result = err;
+			}
+
+			expect(result).toHaveProperty("statusCode");
+			expect(result).toHaveProperty("body");
+			expect(result).toMatchObject({ statusCode: StatusCodeEnum.SUCCESS });
+			expect(typeof result.body).toBe("string");
+			expect(JSON.parse(result.body)).toMatchObject({
+				accessToken: tokenGenerator({ userId: fakeUserId }).accessToken,
+			});
+			expect(userMock.getByEmail).toBeCalled();
+			expect(refreshTokenMock.create).toBeCalled();
+		});
+	});
+
+	describe("Fail", () => {
+		it("invalid user", async () => {
```

`it("should throw an error if user doesn't exists",`